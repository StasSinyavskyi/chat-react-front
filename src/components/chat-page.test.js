/* eslint-env jest */
import renderer from 'react-test-renderer';
import React from 'react';
import ReactDOM from 'react-dom';
import ChatPage from './chat-page';

jest.mock('./sidebar-wrap', () => () => 'SidebarWrap');
jest.mock('./chat-header', () => () => 'Chatheader');
jest.mock('./chat', () => () => 'Chat');
jest.mock('./error-message', () => () => 'ErrorMessage');

// if we have several test we can join them in describe
describe('<ChatPage />', () => {
  const testParam = (
    <ChatPage
      fetchAllChats={() => 'fetchAllChats'}
      fetchMyChats={() => 'fetchMyChats'}
      setActiveChat={() => 'setActiveChat'}
      socketsConnect={() => 'socketsConnect'}
      mountchat={() => 'mountchat'}
      unmountchat={() => 'unmountchat'}
      match={{ params: { chatId: '123' } }}
      active={{ }}
      chats={{ active: {}, my: [], all: [] }}
    />
  );

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(testParam, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('render correctly', () => {
    const tree = renderer
      .create(testParam)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
