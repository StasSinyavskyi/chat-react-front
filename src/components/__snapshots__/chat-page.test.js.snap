// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatPage /> render correctly 1`] = `
<div
  className="ChatPage-root-2"
>
  Chatheader
  SidebarWrap
  Chat
  ErrorMessage
</div>
`;
